gdal/swig/python/scripts/gdal2tiles.py

```diff
@@ -462,65 +467,908 @@ def tilefilename(self, x, y, z):
+                    "Options for generated Google Earth SuperOverlay metadata")
+    g.add_option("-k", "--force-kml", dest='kml', action="store_true",
+                 help=("Generate KML for Google Earth - default for 'geodetic' profile and "
+                       "'raster' in EPSG:4326. For a dataset with different projection use "
+                       "with caution!"))
+    g.add_option("-n", "--no-kml", dest='kml', action="store_false",
+                 help="Avoid automatic generation of KML files for EPSG:4326")
+    g.add_option("-u", "--url", dest='url',
+                 help="URL address where the generated tiles are going to be published")
+    p.add_option_group(g)
+
+    # HTML options
+    g = OptionGroup(p, "Web viewer options",
+                    "Options for generated HTML viewers a la Google Maps")
+    g.add_option("-w", "--webviewer", dest='webviewer', type='choice', choices=webviewer_list,
+                 help="Web viewer to generate (%s) - default 'all'" % ",".join(webviewer_list))
+    g.add_option("-t", "--title", dest='title',
+                 help="Title of the map")
+    g.add_option("-c", "--copyright", dest='copyright',
+                 help="Copyright for the map")
+    g.add_option("-g", "--googlekey", dest='googlekey',
+                 help="Google Maps API key from http://code.google.com/apis/maps/signup.html")
+    g.add_option("-b", "--bingkey", dest='bingkey',
+                 help="Bing Maps API key from https://www.bingmapsportal.com/")
+    p.add_option_group(g)
+
+    p.set_defaults(verbose=False, profile="mercator", kml=False, url='',
+                   webviewer='all', copyright='', resampling='average', resume=False,
+                   googlekey='INSERT_YOUR_KEY_HERE', bingkey='INSERT_YOUR_KEY_HERE',
+                   processes=1)
+
+    return p
+
+
+def process_args(argv):
+    parser = optparse_init()
+    options, args = parser.parse_args(args=argv)
+
+    # Args should be either an input file OR an input file and an output folder
+    if (len(args) == 0):
+        exit_with_error("You need to specify at least an input file as argument to the script")
+    if (len(args) > 2):
+        exit_with_error("Processing of several input files is not supported.",
+                        "Please first use a tool like gdal_vrtmerge.py or gdal_merge.py on the "
+                        "files: gdal_vrtmerge.py -o merged.vrt %s" % " ".join(args))
+
+    input_file = args[0]
+    if not os.path.isfile(input_file):
+        exit_with_error("The provided input file %s does not exist or is not a file" % input_file)
+
+    if len(args) == 2:
+        output_folder = args[1]
+    else:
+        output_folder = os.path.basename(input_file)
+
+    options = options_post_processing(options, input_file, output_folder)
+
+    return input_file, output_folder, options
+
+
+def options_post_processing(options, input_file, output_folder):
+    # Workaround for old versions of GDAL
```

Comment:
I think you can remove such workarounds for old GDAL versions. gdal2tiles is released with a libgdal, so even requiring features only present in trunk would be OK